### PR TITLE
Documentation and repository clean-up

### PR DESCRIPTION
**Pull request by FawazTirmizi**
These commits focus on cleaning up the documentation and repository itself. Notable changes include:
- Changelog has been wiped and made to correspond to this package. It would be useful for the reference to v0.0.1 to point to an actual tag; I don't believe I can add that so if someone else could that would be much appreciated.
- Removed unnecessary references to cortex-m-semihosting from the README.
- Clarified places where ARM Semihosting documentation was referenced.
- General cleaning of the file formatting
- CI has been removed, it was cortex-m-semihosting specific. I'm not familiar with how GitHub CI works, but if there is desire to add CI I can attempt it in a later PR.
- Update the Cargo.toml.

If this gets merged I think the crate is in a state where it could be published to crates.io. I do have some other changes I'd like to make, namely a compile-time architecture check as well as a feature to select the privilege level that the semihosting calls will be initiated from, as many embedded systems nowadays implement at least M/U mode. In my work on Hubris, this has been an issue I've ran into, so I have these features implemented in our fork. I think this would be fairly useful to those who use this crate, so I'd be happy to merge upstream if there's interest.

Pinging  @Disasm for review.

### CODE REVIEW

File: ci/install.sh [hunk 1]
```diff
@@ -1,12 +0,0 @@
-set -euxo pipefail
```
Comment (Disasm) [inline ci/install.sh]: These scripts are used in Travis CI (which is not active I guess), so it would be nice to replace these with Github Actions CI

Comment (FawazTirmizi) [reply ci/install.sh]: I can try and add that, would you like to see it done in this PR?

Comment (Disasm) [reply ci/install.sh]: Thanks! I'd create a separate PR and leave the CI files intact in this PR.

**Comment by FawazTirmizi** [comment]
Reverted the CI removal. Is there anything else that should be done?

**Comment by Disasm** [comment]
No, looks good, thank you!

**Review by Disasm — APPROVED** [review]

**Pull request merged**